File: Cmdlets/ImportCultureAndRegionInfoCommand.vb
Imports System.Globalization
Imports System.Management.Automation

<Cmdlet("Import", "CultureAndRegionInfo")>
Public Class ImportCultureAndRegionInfoCommand
	Inherits CultureAndRegionInfoCommandBase

	<Parameter(Mandatory:=True)>
	Public Property FilePath() As String


	Protected Overrides Sub BeginProcessing()
		Dim builder As CultureAndRegionInfoBuilder = CultureAndRegionInfoBuilder.CreateFromLdml(FilePath)

		CultureAndRegionInfoDictionary.Add(builder.CultureName, builder)
	End Sub
End Class

File: C#/Cmdlets/Cmdlets/ImportCultureAndRegionInfoCommand.cs
using System.Globalization;
using System.Management.Automation;

namespace TIKSN.Cmdlets
{
    [Cmdlet("Import", "CultureAndRegionInfo")]
    public class ImportCultureAndRegionInfoCommand : CultureAndRegionInfoCommandBase
    {
        [Parameter(Mandatory = true)]
        public string FilePath { get; set; }

        protected override void BeginProcessing()
        {
            CultureAndRegionInfoBuilder builder = CultureAndRegionInfoBuilder.CreateFromLdml(FilePath);

            CultureAndRegionInfoDictionary.Add(builder.CultureName, builder);
        }
    }
}